### ISSUE TITLE
Indic Input 2 and 3 tools available for typing in Indian languages are not accessible with NVDA.

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4932]
**Reported by sumandogra on 2015-02-18 10:45**
The Indic Input 2 and 3 tools available for some Indian languages are not accessible with keyboard.  There is no indication on reaching the input tools with keyboard present in the system tray.  One cannot use the phonetics keyboard and like Ramington keyboard  provided by the tools in various Indian languages.  

Indic Input 2 and Indic Input 3 are tools that give users a convenient way of entering text in Indian Languages using the English QWERTY keyboard. User can enter Lingual text in Office Applications, WordPad and Notepad on various Operating Systems. Tools have several common features like on-the-fly help, multiple keyboard layouts and toggling between Languages.                 

The tool supports all popular keyboard layouts of a language.

Pls refer the link:   http://bhashaindia.com/Downloads/Pages/Home.aspx

Steps:
1.  Install the Indic Input tool.
2.  The tool is available in sytem tray.
3.  Use TAB to reach the tool.
4.  The focus does reach to the input indicator.
5.  Now use ARROWS.  Nothing is spoken.  No information is available to the user.  The features are completely inaccessible.

**Comment by ManshulBelani** [comment]
Nothing was being spoken on navigating language layout menu items using arrow keys because when menu items have focus, focused state was not being set, nor is the focused state set for any of the ancestors. This was probably at least one reason NVDA is filtering this out.

To solve this, we created an overlay class in which this filtering was removed and focus was set on these items. However due to this, once an item from these menu items is selected, we are explicitly setting the focus on that item and therefore on continuing object navigation after this, we get unknown objects. 
Is there some way to get the focus back on language bar after some selection is made from these menu items? (On checking the Inspect tool, the hierarchy above language bar also seems to be broken)

How found:	Mouse move (1650,1057)
hwnd=0x00060230 64bit
class="CiceroUIWndFrame" 
style=0x54010000 ex=0x0
ChildId:	1
Interfaces:	IAccIdentity
Impl:	Remote native IAccessible
AnnotationID:	0100008030020600FCFFFFFF01000000
Name:	"English (India)"
Value:	[Error: hr=0x80020003 - Member not found.]
Role:	menu button (0x39)
State:	default (0x100)
Location:	{l:1641, t:1045, w:20, h:26}
Description:	"Input Language Button"
Kbshortcut:	[Error: hr=0x80020003 - Member not found.]
DefAction:	"Press"
Help:	[Error: hr=0x80020003 - Member not found.]
HelpTopic:	""
ChildCount:	0
Window:	0x60230
FirstChild:	[null]
LastChild:	[null]
Next:	"Help" : menu button : default
Previous:	"Language bar" : tool bar : default
Left:	[null]
Up:	[null]
Right:	"Help" : menu button : default
Down:	[null]
Other Props:	Object has no additional properties
Children:	Container has no children
Ancestors:	"Language bar" : tool bar : default
	"Language bar" : window : focusable
	none : client : default,focusable
	none : window : focusable
	none : client : focusable
	none : window : focusable
	"Desktop" : client : focusable
	"Desktop" : window : focusable
	[ No Parent ]

**Comment by bhavyashah** [comment]
@jcsteh @michaelDCurran @feerrenrut @leonardder Any thoughts on the implementation strategy proposed by @ManshulBelani ?

**Comment by Adriani90** [comment]
Did anybody answer already the question from @ManshulBelani? Is this case solved?

**Comment by Adriani90** [comment]
cc: @josephslb

**Comment by ManshulBelani** [comment]
@bhavyashah @Adriani90 I do have an addon ready for this.
I just need some help with a microsoft registry setting and it will be good
to go. The solution is usable without the registry setting also but in this
case you need to manually change some  language settings in your system.
Does anybody have an idea about the registry setting that handle ls the
setting "Use desktop language bar when available" checkbox in the advanced
keyboard settings? If yes please let me know.

**Comment by Adriani90** [comment]
@ManshulBelani for Windows 10 I am not sure if there is a registry setting / group policy. But at least for Windows 8 and 8.1 you can find a really good treat here:
https://www.reddit.com/r/SCCM/comments/3a7dso/xpost_rwindows8_use_the_desktop_language_bar_when/
I hope you can find some useful information for this.

**Comment by Adriani90** [comment]
And here are some reg files which can change the flaoting behavior of the language bar in Windows 10.
https://www.tenforums.com/tutorials/103041-turn-off-language-bar-input-indicator-windows-10-a.html
Look for example to Option 4.

**Comment by lukaszgo1** [comment]
@ManshulBelani wrote:
> 
 The solution is usable without the registry setting also but in this case, you need to manually change some language settings in your system. Does anybody have an idea about the registry setting that handles the setting "Use desktop language bar when available" checkbox in the advanced keyboard settings? If yes please let me know.
>

I am not big fan of my  screen reader messing with the registry. Furthermore what about the portable copies. If the setting mentioned above is needed for this solution it simply should be documented in the user guide.

**Comment by Adriani90** [comment]
@ManshulBelani is this working properly now with NVDA 2024.1 RC1 and Windows 10 or Windows 11?

**Comment by Adriani90** [comment]
No updates from the initial author upon request since many years. I am closing as abandoned. If you are still having this issue, please comment and we can reopen.
